fairscale/nn/data_parallel/fully_sharded_data_parallel.py

```diff
@@ -187,6 +187,7 @@ def __init__(
         self.buffer_dtype = buffer_dtype or self.compute_dtype
         self.move_grads_to_cpu = cpu_offload if move_grads_to_cpu is None else move_grads_to_cpu
         self.bucket_cap_mb = bucket_cap_mb
+        self.gradient_predivide_factor = self.get_gradient_predivide_factor(self.world_size)
```

Comment:
Can we simplify this to something like (not fully tested):
```python
self.gradient_predivide_factor = max(1, int(self.world_size ** 0.5))
self.gradient_postdivide_factor = self.world_size // self.gradient_predivide_factor
assert self.gradient_predivide_factor * self.gradient_postdivide_factor == self.world_size
```

Reply:
this doesn't work nicely if the world_size is not an exact square.
e.g. `x = max(1, int(32 ** 0.5))` => 5
`y = 32 // 5` => 6
` x * y ` ==> 30 and not 32

Reply:
ah of course you are right. Current solution looks good then 😄